exporters/dsexporter/dsexporter_test.go

```diff
@@ -2,45 +2,113 @@ package main
+	check := "prometheus-appstudio-ds"
+	result := GetDataSourceAvailability(datasources, check)
+	if result != 1 {
+		t.Errorf("Expected result 1, but got %f", result)
+	}
 
-	// Simulate collecting metrics and check the exported metric value.
-	metrics := prometheus.DefaultGatherer
-	metricFamilies, err := metrics.Gather()
-	assert.NoError(t, err)
+	check = "prometheus-appstudio-dsexporter"
+	result = GetDataSourceAvailability(datasources, check)
+	if result != 0 {
+		t.Errorf("Expected result 0, but got %f", result)
+	}
+}
 
-	var RequestCountValue float64
-	for _, mf := range metricFamilies {
-		if mf.GetName() == "request_count" {
-			RequestCountValue = mf.GetMetric()[0].GetCounter().GetValue()
-			break
+func TestGetGrafanaResource(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path != "/apis/grafana.integreatly.org/v1beta1/namespaces/appstudio-grafana/grafanas/grafana-oauth" {
+			t.Errorf("Unexpected request path: %s", r.URL.Path)
 		}
+		response := `{"test": "data"}`
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(response))
+	}))
+	defer server.Close()
+
+	config := &rest.Config{
+		Host: server.URL,
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if(err != nil) {
+		t.Fatalf("Error: %v", err)
+	}
+	result := GetGrafanaResource(clientset)
+	expectedResult := map[string]interface{}{"test": "data"}
+	if !reflect.DeepEqual(result, expectedResult) {
+		t.Errorf("Expected %v, but got %v", expectedResult, result)
 	}
+}
 
-	// Check whether the exported metric value is initially 0.
-	assert.Equal(t, float64(0), RequestCountValue)
+func TestGetDataSources(t *testing.T) {
+	grafanaRes := map[string]interface{}{
+		"status": map[string]interface{}{
+			"datasources": []interface{}{"appstudio-ds1", "appstudio-ds2"},
+		},
+	}
 
-	// Increment the requestCounter by calling the Inc method.
-	exporter.requestCounter.Inc()
+	expectedResult := []string{"appstudio-ds1", "appstudio-ds2"}
+	result := GetDataSources(grafanaRes)
+	if !reflect.DeepEqual(result, expectedResult) {
+		t.Errorf("Test-1 failed, Expected %v, but got %v", expectedResult, result)
+	}
 
-	// Collecting metrics again
-	metricFamilies, err = metrics.Gather()
-	assert.NoError(t, err)
+	// empty datasources
+	grafanaRes = map[string]interface{}{
+		"status": map[string]interface{}{
+			"datasources": []interface{}{},
+		},
+	}
 
-	for _, mf := range metricFamilies {
-		if mf.GetName() == "request_count" {
-			RequestCountValue = mf.GetMetric()[0].GetCounter().GetValue()
-			break
-		}
+	expectedResult = []string{}
+	result = GetDataSources(grafanaRes)
+	if !reflect.DeepEqual(result, expectedResult) {
+		t.Errorf("Test-2 failed, Expected %v, but got %v", expectedResult, result)
+	}
+}
+
+func TestMain(t *testing.T) {
+	exporter := prometheus.NewGaugeVec(
```

Comment:
Can't we use the `NewCustomCollector` instead of copy it's content?

Reply:
Updated to use `NewCustomCollector`